youtube_dl/YoutubeDL.py

```diff
@@ -948,7 +953,14 @@ def iter_playlistitems(format):
                 entry_result = self.process_ie_result(entry,
                                                       download=download,
                                                       extra_info=extra)
+
+                entry_result_uploaddate = entry_result.get('upload_date')
+                if entry_result_uploaddate:
+                    if self.params.get('date_playlist_order') in ('desc', 'asc') and entry_result_uploaddate not in self.params.get('daterange'):
```

Comment:
Did you even bother to run your code? `self.params.get('date_playlist_order')` will always be `None` thus `break` will never trigger since [you did not forward it to `params`](https://github.com/rg3/youtube-dl/blob/master/youtube_dl/__init__.py#L310-L429). Even if you did, it **won't work** (repeating this 3rd time already) if `daterange` is an **inner interval** inside `daterange`. In such case if the first/last video (depending on the order) does not belong to `daterange` extraction will stop on it and won't process further items that may belong to `daterange`.